### ISSUE TITLE
statusCheck in the skaffold.yaml is not used without --status-check= flag

### ISSUE THREAD

**Issue by johnmanjiro13** [GoogleContainerTools/skaffold#7178]
### Expected behavior

`skaffold run` runs without status check only setting the `statusCheck` field of the deployment config stanza in the skaffold.yaml to false.

### Actual behavior

The statusCheck field in the skaffold.yaml is used only when value of the `--status-check` flag is nil.
https://github.com/GoogleContainerTools/skaffold/blob/v1.36.1/pkg/skaffold/runner/deployer.go#L53-L60

However, the default value of the flag is true. So we have to use `--status-check=`.
This seems like a strange behavior.

### Information

- Skaffold version: v1.36.1
- Operating system: MacOS Big Sur (11.6)
- Installed via: Homebrew
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta27
kind: Config
metadata:
  name: deployment
build:
  artifacts:
    - image: eu.gcr.io/example/push-events
      docker:
        dockerfile: backend/push-events-svc/Dockerfile
deploy:
  helm:
    releases:
      - name: example
        chartPath: ./k8s/chart
        valuesFiles:
          - ./k8s/values/dev.yaml
  statusCheck: false
```

### Steps to reproduce the behavior

1. use the skaffold.yaml
2. skaffold run # run with status check
3. skaffold run --status-check= # run without status check

**Comment by tomas-mota** [comment]
I'll look into this

**Comment by tomas-mota** [comment]
I was not able to replicate your results. Adding statusCheck: false to the examples/buildpacks/skaffold.yaml worked as expected. Can you try that example too?

**Comment by tomas-mota** [comment]
I have tried cloning the repo from scratch, building and running the example with no statuscheck and it works fine for me. Are you sure that you don't have some global configuration messing things up? Perhaps try clearing out your config.

**Comment by johnmanjiro13** [comment]
I check default value of the status-check flag in the main branch and it is nil.
This issue appears to be fixed in #7089.

I'll wait to release new version. Thank you!

**Comment by johnmanjiro13** [comment]
This issue has been fixed in the v1.37.0 release.
So I close this.
